src/main/kotlin/se/uulm/snowballr/backend/repository/ProjectTableRepo.kt

```diff
@@ -122,4 +165,51 @@ class ProjectTableRepo(
             .andWhere { projectFilter }
             .map { it.toProject() }
     }
+
+    override suspend fun updateProject(request: ProjectOuterClass.Project.Update, activeLocked: Boolean): Project =
+        db.dbQuery {
+            val projectId = parseUUID(request.project.id, EntityType.PROJECT)
+            val fieldMaskPaths = FieldMaskUtil.normalize(request.mask).pathsList.toSet()
+
+            ProjectTable.update({ ProjectTable.id eq projectId }) {
+                applyBasicProjectUpdates(it, request, fieldMaskPaths)
+                if (!activeLocked) {
+                    applyUnlockedProjectUpdates(it, request, fieldMaskPaths)
+                }
+                it[modifiedAt] = OffsetDateTime.now()
+            }
+
+            // Return updated project
+            getProjectByIdOrNull(projectId)
+                ?: throw EntityNotPersistedException(EntityType.PROJECT, projectId.toString())
+        }
+
+    private fun applyBasicProjectUpdates(
+        updateBuilder: UpdateBuilder<*>,
+        request: ProjectOuterClass.Project.Update,
+        paths: Set<String>,
+    ) {
+        if ("project.name" in paths) {
+            updateBuilder[ProjectTable.name] = request.project.name
+        }
+        if ("project.status" in paths) {
+            updateBuilder[ProjectTable.status] = request.project.status
+        }
+    }
+
+    private fun applyUnlockedProjectUpdates(
```

Comment:
What about this name?
```suggestion
    private fun applySlrSettingsUpdates(
```

Reply:
Yes, probably better.